wgpu-core/src/device/global.rs

```diff
@@ -188,11 +188,19 @@ impl Global {
                 hal::BufferUses::empty()
             } else if desc.usage.contains(wgt::BufferUsages::MAP_WRITE) {
                 // buffer is mappable, so we are just doing that at start
+                let snatch_guard = device.snatchable_lock.read();
```

Comment:
should only be takein inside the else block, no need to bother that lock otherwise